client/src/app/metadata-schema-form/metadata-field-types/base-input/base-input.component.spec.ts

```diff
@@ -48,12 +46,27 @@ describe('BaseInputComponent', () => {
   });
 
   it('should initialise attributes based on its property', () => {
-
-    expect(component.helperText).toEqual('Enter in the format: first name,middle name or initial,last name.');
     expect(component.isRequired).toEqual(false);
     expect(component.disabled).toEqual(false);
     expect(component.placeholder).toEqual('John,D,Doe; Jane,,Smith');
     expect(component.label).toEqual('Contact name');
 
   });
+
+  it('should initialise helper text from description and guidelines from schema by default', () => {
+    expect(component.helperText).toEqual('Name of individual who has contributed to the project.<br/><br/>Enter in the format: first name,middle name or initial,last name.');
+  });
+
+  it('should initialise helper text from custom metadata object guideline', () => {
+    fixture = TestBed.createComponent(BaseInputComponent);
+    component = fixture.componentInstance;
+    component.metadata = new Metadata({
+      schema: schema as JsonSchemaProperty,
+      key: 'contact',
+      isRequired: false,
+      guidelines: 'Custom helper text'
```

Comment:
This repeats the definition in the `beforeEach` section, apart from the `guidelines` attributes. What is the reason for this repetition?

Reply:
yeah, I can't seem to update the component's metadata after it's created. I'll try to check more on this.